### PR TITLE
add basic groups boilerplate

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Adding in some basic boilerplate for groups service.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

### CODE REVIEW

File: src/pkg/backup/details/details.go [hunk 1]
```diff
@@ -1025,6 +1026,50 @@ func (i *OneDriveInfo) updateFolder(f *FolderInfo) error {
...
+
+func (i *GroupsInfo) UpdateParentPath(newLocPath *path.Builder) {
+	i.ParentPath = newLocPath.PopFront().String()
+}
+
+func (i *GroupsInfo) uniqueLocation(baseLoc *path.Builder) (*uniqueLoc, error) {
+	return nil, clues.New("not yet implemented")
+}
+
+func (i *GroupsInfo) updateFolder(f *FolderInfo) error {
```
Comment (meain) [inline src/pkg/backup/details/details.go:1065]: We can maybe add it to [`updateFolder`](https://github.com/alcionai/corso/blob/9e655e029cdbd300beddbbeffbb6512fd81e9cf3/src/pkg/backup/details/details.go#L788). Same for `uniqueLocation`.

**Review by meain — APPROVED** [review]
LGTM

File: src/pkg/backup/details/details.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/pkg/backup/details/details.go:1065]: I think I got this in a follow-up PR.

File: src/pkg/backup/details/details.go [hunk 2]
```diff
@@ -1025,6 +1026,50 @@ func (i *OneDriveInfo) updateFolder(f *FolderInfo) error {
 	return updateFolderWithinDrive(OneDriveItem, i.DriveName, i.DriveID, f)
 }
 
+// GroupsInfo describes a groups item
+type GroupsInfo struct {
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/backup/details/details.go:1030]: I'm waffling on that.  `Groups` (and similarly, `Teams`) with the plural is the formal name of the service.  Unlike the others, which are formally named by their singular.  Let's talk this over briefly in the meeting tomorrow to see if there are strong feelings either way, yeah?

**Pull request merged**